pallets/automation-time/src/tests_calculation.rs

```diff
@@ -0,0 +1,212 @@
+// This file is part of OAK Blockchain.
```

Comment:
curious why redo the runtime for tests here. does the existing mock file for automation time not work?

Reply:
I want to change the value of the constant `MaxScheduleSeconds`.

https://github.com/OAK-Foundation/OAK-blockchain/blob/6b5e7c9158028539079f959e4b3f9bae1b68d917/pallets/automation-time/src/tests_calculation.rs#L129

Otherwise, the `ArithmeticError::Overflow` case cannot be tested.

Is there a better way? please give me some advices.

Reply:
why specifically this one test - there's a bunch of overflow errors. are you just trying to test any one of them to make sure one succeeds? If so, to me, this one scenario seems pretty impossible to actually happen, since we define the MaxScheduleSeconds and we won't ever set it to MAX and the current_time_slot shouldn't really be incorrect, so this overflow should never realistically happen. is there a more realistic example that you could test? (i say realistic because it might be easier to replicate). So for example, https://github.com/OAK-Foundation/OAK-blockchain/pull/187/files#diff-e6a13f7032a383e19b77d20043ca46041d94f5e49396812e2de1b2000c301e59R1154 or https://github.com/OAK-Foundation/OAK-blockchain/pull/187/files#diff-e6a13f7032a383e19b77d20043ca46041d94f5e49396812e2de1b2000c301e59R1159 might be easier to mock without having to write a whole new runtime file

but if neither of those work and you can't find any other alternative than to remock the entire runtime file, is there any chance you might be able to write a runtime file without any of the XCM pieces? there's a lot of stuff in this mock runtime that may not necessarily need to be there i think? (or maybe it does, but see if you can rip stuff out since it shouldn't bother your test)

also, couldn't you test with validate_and_schedule_task function rather than the schedule notify extrinsic, since the validate_and_schedule_task function was the one changed?

Reply:
**1. there's a bunch of overflow errors. are you just trying to test any one of them to make sure one succeeds?**

Yes, it's hard to construct the test conditions that trigger it since these cases rarely happen.
In fact this is the only test condition I have constructed.

**2. is there any chance you might be able to write a runtime file without any of the XCM pieces?** 

Currently it doesn't, Origin must implement cumulus_pallet_xcm::Origin.

```
impl pallet_automation_time::Config for Test {
...
	type Origin = Origin;
}
```

Complation error: 
```
error[E0277]: the trait bound `Result<cumulus_pallet_xcm::Origin, tests_calculation::Origin>: From<tests_calculation::Origin>` is not satisfied
   --> pallets/automation-time/src/tests_calculation.rs:134:16
    |
134 |     type Origin = Origin;
    |                   ^^^^^^ the trait `From<tests_calculation::Origin>` is not implemented for `Result<cumulus_pallet_xcm::Origin, tests_calculation::Origin>`
    ```




https://github.com/OAK-Foundation/OAK-blockchain/blob/9ec1400cb72650d8d86667f36cc3982eed2ef99d/pallets/automation-time/src/lib.rs#L228

Reply:
**3. couldn't you test with validate_and_schedule_task function rather than the schedule notify extrinsic?**
 
 Good advice! I re-implemented as you suggested.